File: androidApp/src/main/kotlin/fr/abknative/outgo/android/ui/theme/Type.kt
package fr.abknative.outgo.android.ui.theme

import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.Font
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp
import fr.abknative.outgo.android.R
import fr.abknative.outgo.core.ui.DesignTypography

data class OutgoTypography(
    val title: TextStyle,      // EXTRALARGE / Bold
    val subtitle: TextStyle,   // LARGE / SemiBold
    val body: TextStyle,       // MEDIUM / Regular
    val caption: TextStyle,    // SMALL / Regular
    val label: TextStyle       // EXTRA_SMALL / Medium
)

val Poppins = FontFamily(
    Font(R.font.poppins_regular, FontWeight.Normal),
    Font(R.font.poppins_medium, FontWeight.Medium),
    Font(R.font.poppins_bold, FontWeight.Bold)
)

val OutgoTypographyInstance = OutgoTypography(
    title = TextStyle(
        fontFamily = Poppins,
        fontWeight = FontWeight.Bold,
        fontSize = DesignTypography.EXTRALARGE.sp
    ),
    subtitle = TextStyle(
        fontFamily = Poppins,
        fontWeight = FontWeight.SemiBold,
        fontSize = DesignTypography.LARGE.sp
    ),
    body = TextStyle(
        fontFamily = Poppins,
        fontWeight = FontWeight.Normal,
        fontSize = DesignTypography.MEDIUM.sp
    ),
    caption = TextStyle(
        fontFamily = Poppins,
        fontWeight = FontWeight.Normal,
        fontSize = DesignTypography.SMALL.sp
    ),
    label = TextStyle(
        fontFamily = Poppins,
        fontWeight = FontWeight.Medium,
        fontSize = DesignTypography.EXTRA_SMALL.sp
    )
)

File: iosApp/iosApp/ui/theme/Type.swift
import SwiftUI
import SharedApp

struct OutgoTypography {
    let title: Font
    let subtitle: Font
    let body: Font
    let caption: Font
    let label: Font
}

extension AppTheme {
    static func typography(theme: OutgoTheme) -> OutgoTypography {
        return OutgoTypography(
            title: .custom(theme.fontBold, size: CGFloat(DesignTypography.shared.EXTRALARGE)),
            subtitle: .custom(theme.fontMedium, size: CGFloat(DesignTypography.shared.LARGE)),
            body: .custom(theme.fontRegular, size: CGFloat(DesignTypography.shared.MEDIUM)),
            caption: .custom(theme.fontRegular, size: CGFloat(DesignTypography.shared.SMALL)),
            label: .custom(theme.fontMedium, size: CGFloat(DesignTypography.shared.EXTRA_SMALL))
        )
    }
}
